token translator

LivingTribunal18 opened the pull request:
translation of the code that is in the text file to tokens

LivingTribunal18 committed:
token translator

translation of the code that is in the text file to tokens

LivingTribunal18 committed:
Valid translator to tokens

fixed bugs

arcturus5340 reviewed (changes requested):
You need to create vector<vector<string>> with all parsed code and allocate to tokens. Then write a loop in a loop to go through all the values (commands and words of code) and compile a vector of tokens, which you will compare with templates. See my comments in code

arcturus5340 commented on lang.cpp:
```diff
@@ -45,39 +47,48 @@ int main(){
 
   vector<string> transfer;
-
   string id = "Identifier";
```
You don't need these variables, because You can push constants:
`transfer[i].push_back("Literal")`

arcturus5340 commented on lang.cpp:
```diff
@@ -45,39 +47,48 @@ int main(){
+      if (code_list[i] == ":") {
+          transfer.push_back(op);
+      }else if (code_list[i] == ";" || code_list[i] == "!" || code_list[i] == "close") {
+          break;
```
I don't understand, why do You need to exit the cycle here?

arcturus5340 commented on lang.cpp:
```diff
@@ -45,17 +47,48 @@ int main(){
+          for (int j = 0; j < code_list[i].length(); j++) {
+             if (code_list[i] >= "0" && code_list[i] <= "9"){
+               transfer.push_back(lit);
+               break;
```
Your PL can operate just with single digit numbers? You break loop after first iteration. That's not cool..

arcturus5340 reviewed (changes requested):
Even your code need stylistic refactoring. 72 line of lang.cpp makes me cry blood

LivingTribunal18 committed:
Update lang.cpp

in the vector transfer are placed not variables with the value of tokens, but already integer constants-tokens.
expanded the range of valid numbers from 0 to 9999.

arcturus5340 commented on lang.cpp:
```diff
@@ -59,26 +59,23 @@ int main(){
       }else if (code_list[i] == ";" || code_list[i] == "!" || code_list[i] == "close") {
           break;
       }
           for (int j = 0; j < code_list[i].length(); j++) {
```
You do not use a variable j. Why do you need a loop

arcturus5340 commented on lang.cpp:
```diff
@@ -59,26 +59,23 @@ int main(){
           for (int j = 0; j < code_list[i].length(); j++) {
-             if (code_list[i] >= "0" && code_list[i] <= "9"){
-               transfer.push_back(lit);
+             if (code_list[i] >= "0" && code_list[i] <= "9999"){
```
It looks not very professional

arcturus5340 commented on lang.cpp:
```diff
@@ -59,26 +59,23 @@ int main(){
+               transfer.push_back("Literal");
                break;
             }else {
                break;
```
Double break indicates unprofessionalism. You don't need a loop here

arcturus5340 reviewed (commented):
Too much useless code and my idea from the previous comment has not been implemented

arcturus5340 commented on lang.cpp:
```diff
@@ -59,26 +59,23 @@ int main(){
 */
 
   vector<string> transfer;
```
As I said, you need a vector of vectors. The code is better divided into the vector of commands. And a command on the vector of tokens

arcturus5340 reviewed (changes requested):
You should change the parsing algorithm and use something like linter

LivingTribunal18 committed:
Add files via upload

Removed extra break. Created a new vector of vectors with parsed code.
